gzip.go

```diff
@@ -162,9 +176,18 @@ func MustNewGzipLevelHandler(level int) func(http.Handler) http.Handler {
 // if an invalid gzip compression level is given, so if one can ensure the level
 // is valid, the returned error can be safely ignored.
 func NewGzipLevelHandler(level int) (func(http.Handler) http.Handler, error) {
+	return NewGzipLevelAndMinSize(level, DefaultMinSize)
+}
+
+// NewGzipLevelAndMinSize do as above but let caller specify the minimum size
+// before compression
+func NewGzipLevelAndMinSize(level, askedMinSize int) (func(http.Handler) http.Handler, error) {
 	if level != gzip.DefaultCompression && (level < gzip.BestSpeed || level > gzip.BestCompression) {
 		return nil, fmt.Errorf("invalid compression level requested: %d", level)
 	}
+	if askedMinSize < 0 {
+		return nil, fmt.Errorf("Minimum size must be more than zero")
```

Comment:
Please lowercase the `M` here and I think we're good to go: https://github.com/golang/go/wiki/CodeReviewComments#error-strings